### PR TITLE
Pin hdf5

### PR DESCRIPTION
**Pull request by ocefpaf**
Ping @pelson this should fix `xarray` issue with `libnetcdf` and `hdf5`.

### CODE REVIEW

**Comment by ocefpaf** [comment]
I :heart:  the new `conda-linter-bot` check!

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,25 +10,27 @@ source:
...
         - python
         - setuptools
-        - h5py
+        # This pins to the default channel version as conda-forge's h5py (hdf5) is incompatible with libnetcdf at the moment.
+        - h5py 2.6.0 np{{ np }}py{{ py }}_0
     run:
         - python
-        - h5py
+        # This pins to the default channel version as conda-forge's h5py (hdf5) is incompatible with libnetcdf at the moment.
+        - h5py 2.6.0 np{{ np }}py{{ py }}_0
```
Comment (ocefpaf) [inline recipe/meta.yaml]: I don't know the proper syntax to pin to release `0`!

File: recipe/meta.yaml [hunk 2]
```diff
@@ -10,25 +10,26 @@ source:
...
     build:
         - python
         - setuptools
-        - h5py
     run:
         - python
-        - h5py
+        # This pins to the default channel version as conda-forge's h5py (hdf5) is incompatible with libnetcdf at the moment.
+        - numpy x.x
+        - h5py 2.6.0 np{{ np }}py{{ py }}_0
```
Comment (ocefpaf) [inline recipe/meta.yaml]: I am not a big fan of this solution. @jakirkham, @pelson, and @msarahan does any of you know a better way to pin h5py to build number `0`? (Ideally we would pin to the default channel.)

**Comment by pelson** [comment]
So the problem is with h5py, not this package, no? Can we not just fix that and remove any need to fix it here?

**Comment by jakirkham** [comment]
No, my understanding of the problem is we are using a newer version of HDF5 and a newer version of h5py that is linked to it. So we need to rebuild a bunch of stuff to use it. Why are we not going that route?

**Comment by ocefpaf** [comment]
> So the problem is with h5py, not this package, no? Can we not just fix that and remove any need to fix it here?

This is a terrible workaround and I am not eager to merge this. (Specially now that we are close to our own `netcdf`.)

> No, my understanding of the problem is we are using a newer version of HDF5 and a newer version of h5py that is linked to it. So we need to rebuild a bunch of stuff to use it. Why are we not going that route?

There is no need to re-build `h5netcdf` if we can install our `hdf5` and the future `libnetcdf` together. `h5netcdf` is not "linked," it just calls `h5py`. And our `h5py` is already using the latest `hdf5`...

**Pull request closed without merging**